### PR TITLE
add getchannelfeed

### PR DESCRIPTION
**Pull request by bustawei**

### CODE REVIEW

File: apps/meow/src/lib/storage.ts [hunk 1]
```diff
@@ -372,9 +372,32 @@ export class MStore extends ZStore {
...
+    }).collect();
+    for (const m of messages) {
+      const msg = m.payload.value as types.ZChainMessage;
+      const msgValue = await decode(msg.message,password)
+      channelFeed.push([msg,msgValue])
+    }
+    return channelFeed;
+
+  }
   async displayChannelFeed(channel: string, n: number): Promise<void> {
```
Comment (ratik21) [inline apps/meow/src/lib/storage.ts]:
```suggestion
  }
  
  async displayChannelFeed(channel: string, n: number): Promise<void> {
```

File: apps/meow/src/lib/storage.ts [hunk 2]
```diff
@@ -372,9 +372,32 @@ export class MStore extends ZStore {
...
+    }
+
+    await channelDB.load(n); // load last "n" messages to memory
+    const messages = channelDB.iterator({
+      limit: n, reverse: true
+    }).collect();
+    for (const m of messages) {
+      const msg = m.payload.value as types.ZChainMessage;
+      const msgValue = await decode(msg.message,password)
+      channelFeed.push([msg,msgValue])
```
Comment (ratik21) [inline apps/meow/src/lib/storage.ts:397]: Please return the msg as an object, in an array of messages. I think decryption should happen at the "printing"/"display" level. 

You should simply return `[ { ..m1 }, { ..m2 }, .. ,{ ..mN } ]` here

File: apps/meow/src/lib/storage.ts [hunk 3]
```diff
@@ -390,17 +413,13 @@ export class MStore extends ZStore {
...
     console.log(chalk.cyanBright(`Last ${n} messages published on ${channel}`));
     for (const m of messages) {
-      const msg = m.payload.value as types.ZChainMessage;
+      const msg = m[0];
       console.log(`${chalk.green('>')} `, {
-        ...msg,
-        message: await decode(msg.message, password)
+        msg,
+        message: m[1]
       });
```
Comment (ratik21) [inline apps/meow/src/lib/storage.ts:426]: Could you please explain why we're doing this? `m[0]` or `m[1]` is not very verbose.

Comment (ratik21) [reply apps/meow/src/lib/storage.ts:426]: Please check above comment.

File: apps/meow/src/lib/storage.ts (hunk 2 above)
Comment (bustawei) [reply apps/meow/src/lib/storage.ts:397]: Password won't be passed to every app that will use meow-app . like in zScreen messages should be received decoded

File: apps/meow/src/lib/storage.ts (hunk 3 above)
Comment (bustawei) [reply apps/meow/src/lib/storage.ts:426]: this is just printing part , m[0] is the whole zchainMessageObject and m[1] is the decoded message

**Pull request closed without merging**